server/src/main/java/org/elasticsearch/action/support/DefaultShardOperationFailedException.java

```diff
@@ -129,6 +143,15 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endObject();
         }
         return builder;
+    }
+
+    public static DefaultShardOperationFailedException fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
 
+    private static final class Fields {
```

Comment:
let's keep these constants as top-level fields, no need for the `Fields` inner class. This is something we used to do in the past that we want to stop doing it and remove in the future.